inc/ThirdParty/Themes/Divi.php

```diff
@@ -214,4 +218,109 @@ public function disable_dynamic_css_on_rucss() {
 	public function remove_assets_generated() {
 		remove_all_actions( 'et_dynamic_late_assets_generated' );
 	}
+
+	/**
+	 * Get layout IDs for a template.
+	 *
+	 * @param int $template_post_id Template post ID.
+	 *
+	 * @return array
+	 */
+	private function get_layout_ids( $template_post_id ) {
+		$allowed_post_types = [
+			'et_header_layout',
+			'et_footer_layout',
+			'et_body_layout',
+		];
+		$current_post_type  = get_post_type( $template_post_id );
+
+		if ( ! in_array( $current_post_type, $allowed_post_types, true ) ) {
+			return [];
+		}
+
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery
+		return (array) $wpdb->get_col(
+			$wpdb->prepare(
+				'SELECT post_id from ' . $wpdb->postmeta . ' WHERE meta_key = %s AND meta_value = %d',
+				'_' . $current_post_type . '_id',
+				$template_post_id
+			)
+		);
+	}
+
+	/**
+	 * Save template handler.
+	 *
+	 * @param int $template_post_id Template post ID.
+	 *
+	 * @return void
+	 */
+	public function handle_save_template( $template_post_id ) {
+		/**
+		 * Filters Bypassing saving template functionality.
+		 *
+		 * @param bool $bypass Bypass save template functionality.
+		 * @param int  $template_post_id Currently saved template post id.
+		 */
+		if ( apply_filters( 'rocket_divi_bypass_save_template', empty( $layout_post_ids ), $template_post_id ) ) {
+			return;
+		}
+
+		$layout_post_ids = $this->get_layout_ids( $template_post_id );
+		if ( empty( $layout_post_ids ) ) {
+			return;
+		}
+
+		foreach ( $layout_post_ids as $layout_post_id ) {
+			if ( 'publish' !== get_post_status( $layout_post_id ) ) {
+				continue;
+			}
+
+			$used_on = get_post_meta( $layout_post_id, '_et_use_on' );
+			if ( empty( $used_on ) ) {
+				continue;
+			}
+
+			set_transient( 'rocket_divi_notice', true );
+			return;
+		}
+	}
+
+	/**
+	 * Admin notices handler.
+	 *
+	 * @return void
+	 */
+	public function handle_divi_admin_notice() {
```

Comment:
Aren't we missing a check for RUCSS enabled?

Also, the notice will be displayed again if reloading the page, since the transient is not deleted until `rocket_after_clean_used_css`